sympy/solvers/ode/single.py

```diff
@@ -563,7 +563,7 @@ def _verify(self, fx) -> bool:
             factor = exp(Integral(factor, integration_variable))
             m *= factor
             n *= factor
-            self._wilds_match[P] = m.subs(y, fx)
+            self._wilds_match[P] = m.subs(y, 0)
```

Comment:
0 is not a good choice if factor has a singularity at y=0.

Reply:
This is why the ODE solvers generally avoid typical definite integral type solutions you might find in textbooks. They require that the integration bounds be chosen in some "nice" region of the function's domain, but we can't do that for arbitrary expressions. Instead we use definite integrals and "substituted" integrals (like `integrate(f(x), (x, a))`) and let this complexity live in the implicit restrictions on the arbitrary constants. The general solution in the [documentation](https://docs.sympy.org/latest/modules/solvers/ode.html#st-exact) is based on `x0` and `y0`, but those have to be chosen based on the specific ODE.